src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copy = { ...state };
+  const stateArr = [];
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
```

Comment:
the same as in previous task
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch